Frameworks/CoreText/CTLine.mm

```diff
@@ -240,7 +240,10 @@ void _CTLineDraw(CTLineRef lineRef, CGContextRef ctx, bool adjustTextPosition) {
             CGContextSetTextPosition(ctx, curTextPos.x + curRun->_relativeXOffset, curTextPos.y);
         }
 
-        _CTRunDraw(static_cast<CTRunRef>(curRun), ctx, CFRange{}, false);
+        // Get height of the line so we draw with the correct baseline for each run
+        CGFloat ascent, descent;
+        CTLineGetTypographicBounds(lineRef, &ascent, &descent, nullptr);
+        _CTRunDraw(static_cast<CTRunRef>(curRun), ctx, CFRange{}, false, ascent - descent);
```

Comment:
ascent-descent because descent is negative, right?  Would be useful to add a note here for future.